grpc/worker/worker.proto

```diff
@@ -6,6 +6,8 @@ service Worker {
     rpc sync_from_upstream (SyncRequest) returns (SyncResponse);
     rpc register_accepted (RegisterResponse) returns (RegisterAck);
     rpc health_check (HealthCheckRequest) returns (HealthCheckResponse);
+    rpc get_providers (google.protobuf.Empty) returns (ProvidersResponse);
+    rpc add_resource (AddResourceRequest) returns (AddResourceResponse);
```

Comment:
Firstly, what you need to code is interface from Front-end to Supervisor ,BUT in grpc, it about interface form Supervisor to Worker. 
Secondly, there is no def of resource in code, it just a concept rather than actual char[] or Class. In Class ProviderInfo, ProviderInfo.upstream_url can get these resource.
Thirdly, you don't def AddResourceRequest, and there is not define for it in whole Kagami